### PR TITLE
[Xedra Evolved] Misc

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Mods "[Xedra Evolved] Miscellaneous fixes"

#### Describe the solution

<details>
  <summary>Made a foundation for spell loot groups</summary>

Each class has his own loot group, that describe the items of each tier (from tier 0 to tier 4, 5 tiers in summary). For example, `xe_dreamer_spell_tier2`
Then this list of items goes into level above, into group, that already can go into the map, and contain a spells of the tier_level, and also contain spells of higher and lover tier with smaller probability - it allow us to spawn tier 3 spells in tier 2 group. For example, `xe_dreamer_spell_group2`
After this it goes into the biggest class group, that contain all spells of the class, and can be used for locations, related to this class, and has `low`, `mid` and `high` types, with different probabilities of spell groups. For example, `xe_dreamer_spells_mid`
At last, all of them goes into group, that contain all spells in game, and can be used, when the location is not class related. For example, `xe_all_spells_mid`

noname

It looks complicated, but it's not really

</details>

Turn #61913 into game, with tweaked stats
Edited blood boiling spell to be more melee oriented, but decrease the overall damage, because good lord, 250 damage per 10 second at level 15 is too much
Moved weakening spell into eater file
Fixed stamina spell giving speed buff - now it gives stamina effect, as intended

#### Testing

The game works fine, loot generates correctly, spells casts as intended

#### Additional context

I am surprised the game can handle empty loot groups without any issues

### CODE REVIEW

**Review by Maleclypse — APPROVED** [review]
Thank you for doing this!

File: data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
+[
+  {
+    "type": "item_group",
+    "id": "xe_classless_spells_low",
+    "//": "Overall group for classless spells, if you don't need a precise tier items. Groups are used in mapgen",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json:5]:
```suggestion
    "//": "Overall group for classless spells, if you don't need a precise tier of items. Groups are used in mapgen",
```

File: data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      { "group": "xe_classless_spell_group1", "prob": 12 },
+      { "group": "xe_classless_spell_group2", "prob": 25 },
+      { "group": "xe_classless_spell_group3", "prob": 100 },
+      { "group": "xe_classless_spell_group4", "prob": 50 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "id": "xe_classless_spell_group0",
+    "//": "Each group contain artifacts of his tier, plus spells of +1 and -1 tier with lower probability. Groups are used in mapgen",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json:44]:
```suggestion
    "//": "Tier 0 spell artifacts, plus spells of first tier with lower probability. Groups are used in mapgen",
```
No negative tier artifacts for you :p

File: data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    "entries": [
+      { "group": "xe_classless_spell_tier0", "prob": 50 },
+      { "group": "xe_classless_spell_tier1", "prob": 100 },
+      { "group": "xe_classless_spell_tier2", "prob": 50 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "id": "xe_classless_spell_group2",
+    "//": "each group contain artifacts of his tier, plus spells of +1 and -1 tier with lower probability. group is used in mapgen",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json:61]:
```suggestion
    "//": "Tier 2 spell artifacts, plus spells of +1 and -1 tier with lower probability. group is used in mapgen",
```

File: data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    "entries": [
+      { "group": "xe_classless_spell_tier1", "prob": 50 },
+      { "group": "xe_classless_spell_tier2", "prob": 100 },
+      { "group": "xe_classless_spell_tier3", "prob": 50 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "id": "xe_classless_spell_group3",
+    "//": "each group contain artifacts of his tier, plus spells of +1 and -1 tier with lower probability. group is used in mapgen",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json:72]:
```suggestion
    "//": "Tier 3 spell artifacts, plus spells of +1 and -1 tier with lower probability. group is used in mapgen",
```

File: data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  {
+    "type": "item_group",
+    "id": "xe_classless_spell_group0",
+    "//": "Each group contain artifacts of his tier, plus spells of +1 and -1 tier with lower probability. Groups are used in mapgen",
+    "subtype": "distribution",
+    "entries": [ { "group": "xe_classless_spell_tier0", "prob": 100 }, { "group": "xe_classless_spell_tier1", "prob": 50 } ]
+  },
+  {
+    "type": "item_group",
+    "id": "xe_classless_spell_group1",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json:50]:
```suggestion
    "id": "xe_classless_spell_group1",
    "//": "Tier 1 spell artifacts, plus spells of +1 and -1 tier with lower probability. Groups are used in mapgen",
```

File: data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    "entries": [
+      { "group": "xe_classless_spell_tier2", "prob": 50 },
+      { "group": "xe_classless_spell_tier3", "prob": 100 },
+      { "group": "xe_classless_spell_tier4", "prob": 50 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "id": "xe_classless_spell_group4",
+    "//": "each group contain artifacts of his tier, plus spells of +1 and -1 tier with lower probability. group is used in mapgen",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/itemgroups/spell_artifacts/classless.json:83]:
```suggestion
    "//": "Tier 4 spell artifacts, plus Tier 3 spell artifacts with lower probability. group is used in mapgen",
```

File: data/mods/Xedra_Evolved/itemgroups/spell_artifacts/dreamer.json [hunk 7]
```diff
@@ -0,0 +1,114 @@
+[
+  {
+    "type": "item_group",
+    "id": "xe_dreamer_spells_low",
+    "//": "Overall group for dreamer spells, if you don't need a precise tier items. Groups are used in mapgen",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/itemgroups/spell_artifacts/dreamer.json:5]:
```suggestion
    "//": "Overall group for dreamer spells, if you don't need a precise tier of items. Groups are used in mapgen",
```

File: data/mods/Xedra_Evolved/items/spell_learning_items/spell_learning_items_eater.json [hunk 8]
```diff
@@ -74,6 +74,21 @@
     "color": "white_magenta",
     "use_action": { "type": "learn_spell", "spells": [ "self_healing" ] }
   },
+  {
+    "id": "strange_coffee",
+    "type": "GENERIC",
+    "category": "artifacts",
+    "name": { "str": "strange cup with coffee", "str_pl": "strange cups with coffee" },
+    "description": "A small jar with coffee, moving in weird patterns with no apparent cause.  After watching the swirls for some time you are sure that it loops back on it's movements, like some sort of message.",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/items/spell_learning_items/spell_learning_items_eater.json:82]:
```suggestion
    "name": { "str": "strange cup of coffee", "str_pl": "strange cups of coffee" },
    "description": "A small jar with coffee, moving in weird patterns with no apparent cause.  After watching the swirls for some time, you are sure that it loops back on its movements, like some sort of message.",
```

**Review by TheShadowFerret — COMMENTED** [review]
I only did one itemgroup's descriptions since there's so many :p

**Pull request merged**